tests/integration/cmd_dev_debug_test.go

```diff
@@ -408,4 +409,327 @@ var _ = Describe("odo dev debug command tests", func() {
 			})
 		}))
 	}
+
+	// More details on https://github.com/devfile/api/issues/852#issuecomment-1211928487
+	for _, podman := range []bool{false, true} {
+		podman := podman
+		When("starting with Devfile with autoBuild or deployByDefault components",
+			helper.LabelPodmanIf(podman, func() {
+				BeforeEach(func() {
+					helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
+					helper.CopyExample(filepath.Join("source", "nodejs", "Dockerfile"), filepath.Join(commonVar.Context, "Dockerfile"))
+					helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-autobuild-deploybydefault.yaml"),
+						filepath.Join(commonVar.Context, "devfile.yaml"),
+						helper.DevfileMetadataNameSetter(cmpName))
+				})
+
+				When("running odo dev with some components not referenced in the Devfile", func() {
+					var devSession helper.DevSession
+					var stdout, stderr string
+
+					BeforeEach(func() {
+						var bOut, bErr []byte
+						var err error
+						var envvars []string
+						if podman {
+							envvars = append(envvars, "ODO_PUSH_IMAGES=false")
+						} else {
+							envvars = append(envvars, "PODMAN_CMD=echo")
+						}
+						args := []string{"--debug"}
+						if podman {
+							args = append(args, "--forward-localhost")
+						}
+						devSession, bOut, bErr, _, err = helper.StartDevMode(helper.DevSessionOpts{
+							CmdlineArgs: args,
+							EnvVars:     envvars,
+							RunOnPodman: podman,
+						})
+						Expect(err).ShouldNot(HaveOccurred())
+						stdout = string(bOut)
+						stderr = string(bErr)
+					})
+
+					AfterEach(func() {
+						devSession.Stop()
+						if podman {
+							devSession.WaitEnd()
+						}
+					})
+
+					It("should create the appropriate resources", func() {
+						if podman {
+							By("skipping Kubernetes/OpenShift components that would have been created automatically", func() {
+								linesErr, _ := helper.ExtractLines(stderr)
+								var skipped []string
+								for _, l := range linesErr {
+									if strings.Contains(l, "Kubernetes components are not supported on Podman. Skipping:") {
+										sl := strings.SplitN(l, ": ", 2)
+										if len(sl) < 2 {
+											break
+										}
+										for _, s := range strings.Split(sl[1], ", ") {
+											skipped = append(skipped, strings.TrimSuffix(s, "."))
+										}
+										break
+									}
+								}
```

Comment:
I remember I tried to do something similar using `strings.Trim{Prefix,Suffix}`, but it didn't pass because of the warning icon displayed in the beginning for this line, e.g.:
```
 ⚠  Kubernetes components are not supported on Podman. Skipping: k8s-deploybydefault-true-and-not-referenced, k8s-deploybydefault-not-set-and-not-referenced, ocp-deploybydefault-true-and-referenced, ocp-d
eploybydefault-true-and-not-referenced, ocp-deploybydefault-not-set-and-not-referenced, k8s-deploybydefault-true-and-referenced.
```

Anyways, that's a good point - I'll simplify this a little bit.

Reply:
See [`a85e969` (#6720)](https://github.com/redhat-developer/odo/pull/6720/commits/a85e96902d298cb0a2002e8f72f497202be1a9e2)